src/async-wrap.cc

```diff
@@ -448,6 +452,56 @@ static void DisablePromiseHook(const FunctionCallbackInfo<Value>& args) {
   }, static_cast<void*>(env));
 }
 
+static void AsyncTaskScheduled(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsString())
+    return env->ThrowTypeError("first argument must be a string");
+  if (!args[1]->IsNumber())
+    return env->ThrowTypeError("second argument must be a number");
+  if (!args[2]->IsBoolean())
+    return env->ThrowTypeError("third argument must be a boolean");
+
+  Local<String> taskName = args[0].As<String>();
+  String::Value taskNameValue(taskName);
+  StringView taskNameView(*taskNameValue, taskNameValue.length());
+
+  intptr_t taskId = args[1].As<Integer>()->Value();
+  void* task = reinterpret_cast<void*>(taskId);
+  bool recurring = args[2].As<Boolean>()->Value();
+
+  env->inspector_agent()->AsyncTaskScheduled(taskNameView, task, recurring);
+}
+
+static void AsyncTaskCanceled(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsNumber())
+    return env->ThrowTypeError("first argument must be a number");
+
+  intptr_t taskId = args[0].As<Integer>()->Value();
+  env->inspector_agent()->AsyncTaskCanceled(reinterpret_cast<void*>(taskId));
+}
+
+static void AsyncTaskStarted(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsNumber())
+    return env->ThrowTypeError("first argument must be a number");
+
+  intptr_t taskId = args[0].As<Integer>()->Value();
+  env->inspector_agent()->AsyncTaskStarted(reinterpret_cast<void*>(taskId));
+}
+
+static void AsyncTaskFinished(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsNumber())
+    return env->ThrowTypeError("first argument must be a number");
+
+  intptr_t taskId = args[0].As<Integer>()->Value();
+  env->inspector_agent()->AsyncTaskFinished(reinterpret_cast<void*>(taskId));
+}
```

Comment:
> You could DRY these a little by using a template function

Good point. I am not sure if I understand your idea correctly, I was able to DRY these functions with a helper function that does not require any template:

```c++
static void invokeAsyncTaskFnWithId(void (Agent::*asyncTaskFn)(void*),
                                    const FunctionCallbackInfo<Value>& args);
```

Strangely enough, now that I moved this code from `src/async-wrap.cc` to `src/inspector_agent.cc`, ` ./configure --without-inspector` seems to work fine even with not `#if HAVE_INSPECTOR` guard. Perhaps `src/inspector_agent.cc` is excluded from compilation? Having said that, I'll probably need to modify the javascript part.